crates/bevy_transform/src/systems.rs

```diff
@@ -336,36 +329,4 @@ mod test {
             );
         }
     }
-    #[test]
-    #[should_panic]
-    fn panic_when_hierarchy_cycle() {
```

Comment:
I actually could not come up with a way to generate a cycle with the available commands. I think @DJMcNab added this test. I don't think it's possible to make a hierarchy like this with that change, at least not without some real jank `std::mem::swap` usage.

Reply:
`std::mem::swap` or using `EntityMut::remove` to get ownership of a `Children`/`Parent` and inserting it on an entity could create a cycle for the test. It's a bit janky but given its possible to do we should keep testing it until we can make it impossible

Reply:
Added the test back in.